### PR TITLE
fix: Move `devices` into module under `vmm`

### PR DESCRIPTION
**Pull request by JonathanWoollett-Light**
## Changes

Moved `devices` into module under `vmm`

## Reason

See https://github.com/firecracker-microvm/firecracker/pull/3079

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [x] If a specific issue led to this PR, this PR closes the issue.
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] API changes follow the [Runbook for Firecracker API changes][2].
- [x] User-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.
- [x] New `TODO`s link to an issue.
- [x] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [x] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: src/vmm/Cargo.toml [hunk 1]
```diff
@@ -24,17 +24,22 @@ versionize_derive = "0.1.5"
 vm-allocator = "0.1.0"
 vm-superio = "0.7.0"
 vm-fdt = "0.2.0"
+timerfd = "1.2.0"
+rand = "0.8.5"
```
Comment (zulinx86) [inline src/vmm/Cargo.toml:28]: nit: It seems to be arranged in dictionary order.

File: src/vmm/Cargo.toml [hunk 2]
```diff
@@ -24,17 +24,22 @@ versionize_derive = "0.1.5"
...
+dumbo = { path = "../dumbo" }
 logger = { path = "../logger" }
 mmds = { path = "../mmds" }
 rate_limiter = { path = "../rate_limiter" }
 seccompiler = { path = "../seccompiler" }
 snapshot = { path = "../snapshot"}
 utils = { path = "../utils" }
 virtio_gen = { path = "../virtio_gen" }
+io_uring = { path = "../io_uring" }
+net_gen = { path = "../net_gen" }
```
Comment (zulinx86) [inline src/vmm/Cargo.toml:39]: nit: same for arrangement

File: src/vmm/Cargo.toml [hunk 3]
```diff
@@ -24,17 +24,22 @@ versionize_derive = "0.1.5"
...
 
 [dev-dependencies]
+proptest = { version = "1.0.0", default-features = false, features = ["std"] }
```
Comment (zulinx86) [inline src/vmm/Cargo.toml:42]: nit: same for arrangement

File: src/vmm/tests/devices.rs [hunk 4]
```diff
@@ -3,21 +3,50 @@
 
 #![allow(clippy::undocumented_unsafe_blocks)]
 
-mod serial_utils;
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
```
Comment (zulinx86) [inline src/vmm/tests/devices.rs:7]: This copyright should be removed.

**Review by zulinx86 — CHANGES_REQUESTED** [review]

File: src/vmm/tests/devices.rs (hunk 4 above)
Comment (JonathanWoollett-Light) [reply src/vmm/tests/devices.rs:7]: Fixed.

File: src/vmm/Cargo.toml (hunk 2 above)
Comment (JonathanWoollett-Light) [reply src/vmm/Cargo.toml:39]: Fixed.

File: src/vmm/Cargo.toml (hunk 1 above)
Comment (JonathanWoollett-Light) [reply src/vmm/Cargo.toml:28]: Fixed.

**Review by zulinx86 — APPROVED** [review]

File: src/vmm/Cargo.toml (hunk 1 above)
Comment (JonathanWoollett-Light) [reply src/vmm/Cargo.toml]: Created https://github.com/firecracker-microvm/firecracker/pull/3701 to cover this.

**Review by ShadowCurse — APPROVED** [review]

**Pull request merged**